flutter/lib/src/navigation/navigation_breadcrumb.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:flutter/material.dart';
+import 'package:sentry/sentry.dart';
+
+/// This class makes it easier to record breadcrumbs for navigation events by
+/// accepting Flutters [RouteSettings](https://api.flutter.dev/flutter/widgets/RouteSettings-class.html).
+///
+/// See also:
+///   - https://flutter.dev/docs/cookbook/navigation/navigate-with-arguments
+class NavigationBreadcrumb extends Breadcrumb {
```

Comment:
I like the idea, but I think having a class for each type of breadcrumb is a bit overkill,
I'd be in to expose a nice Factory or named constructor in the Breadcrumb class to take parameters instead of setting values manually, See this on Java:

https://github.com/getsentry/sentry-java/blob/3fad970bef0675f63fcfe63cd2ff871b62743ed6/sentry/src/main/java/io/sentry/Breadcrumb.java#L46-L60

Reply:
`Breadcrumb` is a class of sentry-dart, so I can't add a constructor or factory with `RouteSettings` parameters without adding a Flutter dependency in sentry-dart. So this was my next best idea.

Reply:
yeah I do get that, but I'd not pass the whole `RouteSettings` but rather the values (as parameters), so if another Router FW is gonna be integrated, the facade is already available thru the `Breadcrumb` class.

Reply:
I would like to keep it as it is, because it is pretty Flutter specific and it makes the testing easier.

Also in future versions one could easily add more Flutter specific information. [`PageRoute`](https://api.flutter.dev/flutter/widgets/PageRoute-class.html) and co have much more information, which could potentially be useful. For example the name of the Route itself (eg. MaterialPageRoute, CupertinoPageRoute...) or if the navigation action covers the whole screen or if it is a dialog, bottom sheet or something else.

Reply:
so the code looks good, but we are coupling a class called `NavigationBreadcrumb` which is a pretty generic name for any navigation FW/3rd party lib (to the RouteObserver lib.)

I'd suggest moving all the code to the `SentryNavigatorObserver` class which is pretty much what is used for, don't get me wrong, the abstraction does make sense, but for an SDK that could be integrated with other 3rd parties not (IMO), as the `NavigationBreadcrumb` takes a `RouteSettings`.

so I see 2 alternatives, either we rename `NavigationBreadcrumb` to something more specific related to the `RouteObserver` or we move this logic to the `_addBreadcrumb` class, really your call, happy with both.